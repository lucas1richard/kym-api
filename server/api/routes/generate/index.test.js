const { expect, use } = require('chai');
const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot');

describe('generate routes', () => {
  use(jestSnapshotPlugin());
  const routePrefix = '/api/generate';

  describe(`GET ${routePrefix}/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('gets a random food', async () => {
      const res = await globals.agent.getWithToken(`${routePrefix}/v1`);
      expect(res.statusCode).equals(200);
      expect(Number.isInteger(res.body.id)).to.be.true;
    });
  });
  describe(`GET ${routePrefix}/calculate/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('calculates for goals', async () => {
      const goals = 'proteinGoal=20&carbGoal=30&fatGoal=10';
      const res = await globals.agent.getWithToken(`${routePrefix}/calculate/v1?${goals}`);
      expect(res.statusCode).equals(200);
      expect(Array.isArray(res.body)).to.be.true;
      res.body.forEach((item) => {
        expect(Array.isArray(item.foods)).to.be.true;
        expect(item.weight).to.be.ok;
        expect(item.macros).to.be.ok;
      });
    });

    it('validates goals', async () => {
      const goals = '';
      const res = await globals.agent.getWithToken(`${routePrefix}/calculate/v1?${goals}`);
      expect(res.statusCode).equals(400);
      expect(res.body).toMatchSnapshot();
    });
  });
});
