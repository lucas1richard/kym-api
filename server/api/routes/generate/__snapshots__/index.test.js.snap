// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate routes GET /api/generate/calculate/v1 validates goals 1`] = `
Array [
  Object {
    "context": Object {
      "label": "value",
    },
    "message": "GOALS_REQUIRED",
    "path": Array [],
    "type": "override",
  },
]
`;
